app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -524,45 +562,79 @@ public void verifyLocationPermissions(Activity activity) {
-        }else{
+        } else {
             permissionDenied = false;
         }
     }
 
-    /*
-    // get the selected dropdown list value
-    public void addListenerOnSubmitButton() {
+    public void addItemsOnEmojiScroller() throws IllegalAccessException {
+        Field[] fields = R.drawable.class.getFields();
+        List<Integer> drawables = new ArrayList<Integer>();
+        for (Field field : fields) {
+            // Take only those with name starting with "emoji"
+            if (field.getName().startsWith("emoji")) {
+                drawables.add(field.getInt(null));
+            }
+        }
 
-        //emotionalStateSpinner = (Spinner) findViewById(R.id.emotional_state_spinner);
-        socialSituationSpinner = (Spinner) findViewById(R.id.social_event_spinner);
-        //Button submitButton = (Button) findViewById(R.id.post_mood_button);
-
-        submitButton.setOnClickListener(new View.OnClickListener() {
-
-            @Override
-            public void onClick(View v) {
-
-                emotionalState = String.valueOf(emotionalStateSpinner.getSelectedItem());
-                socialSit = String.valueOf(socialSituationSpinner.getSelectedItem());
-
-                try {
-                    submitMood();
-                    if(editflag) {
-                        Toast.makeText(getApplicationContext(), "Successfully updated!", Toast.LENGTH_SHORT).show();
-                    }
-                    else{
-                        Toast.makeText(getApplicationContext(), "Successfully posted!", Toast.LENGTH_SHORT).show();
-                    }
-                    finish();
-                } catch (DescriptionTooLongException e) {
-                    Toast.makeText(getApplicationContext(), "Your description is too long.", Toast.LENGTH_SHORT).show();
+        emojiList = (LinearLayout) findViewById(R.id.emojiList);
+        for (int i = 1; i <= drawables.size(); i++) {
+
+            LinearLayout emojiLayout = new LinearLayout(this);
+            LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+            params.gravity = Gravity.CENTER_VERTICAL;
+            emojiLayout.setPadding(0,0,20,0);
+            emojiLayout.setOrientation(LinearLayout.VERTICAL);
+            emojiLayout.setTag(i); // This is crucial for grabbing the "index" of the clicked emoji later
+
+            emojiButton = new ImageButton(getApplicationContext());
+            emojiButton.setAdjustViewBounds(true);
+            emojiButton.setMaxHeight(150);
+            emojiButton.setMaxWidth(150);
+            emojiButton.setBackgroundResource(R.color.white);
+            emojiButton.setImageResource(getApplicationContext().getResources().getIdentifier("emoji" + i, "drawable", getApplicationContext().getPackageName()));
+            emojiButton.setPadding(0,0,0,0);
+            emojiLayout.addView(emojiButton);
```

Comment:
All of this functionality builds the ImageView, places the correct emoji image into it, resizes the icon to fit the desired aspect ratio, and finally pushes the ImageView to the upper portion of the LinearLayout we just created. The Textview that we're about to create will occupy its lower segment.